src/rcms/fm/app/level1/HCALlevelOneEventHandler.java

```diff
@@ -1741,4 +1750,35 @@ public void FillMaskedResources(){
       }
     }
   }
+
+  public void checkMaskedappsFormat() throws UserActionException{
+    StringT runkeyName                 = (StringT) functionManager.getHCALparameterSet().get("CFGSNIPPET_KEY_SELECTED").getValue();
+    MapT<MapT<StringT>> LocalRunKeyMap = (MapT<MapT<StringT>>)functionManager.getHCALparameterSet().get("AVAILABLE_RUN_CONFIGS").getValue();
+
+    if (LocalRunKeyMap.get(runkeyName).get(new StringT("maskedapps"))!=null){
+      String[] maskedapps         = LocalRunKeyMap.get(runkeyName).get(new StringT("maskedapps")).getString().split("\\|");
+      String errorMessage         = "";
+      for (String app:maskedapps){
+        String[] appArray = app.split("\\_");
+        if (appArray.length != 2 || stringToInt(appArray[0]) || !(stringToInt(appArray[1]))){
+          errorMessage = errorMessage + " " + app;
+        }
+      }
+      if (errorMessage != ""){
+        throw new UserActionException("Runkey " + runkeyName +" maskedapps incorrectly formated:" + errorMessage); 
+      }
+    }
+  }
+
+ public static boolean stringToInt(String s){
```

Comment:
Could we give a more descriptive name to this method? I would expect a function named `stringToInt` to behave like the c++ `std::atoi` that takes a string and returns an int. Maybe a better name would be something like `isValidInstanceNumber`?

The change would make reading line 1763 a bit more intuitive... at a quick glance one might think you are trying to cast `appArray[0]` (the app's class name) to an integer, whereas I guess what this is doing is trying to make sure that the class name isn't accidentally an integer (though I wonder if there is some more obvious way to check whether the class name of the masked app is a valid class name).